tests/unit/test_distro.py

```diff
@@ -346,6 +346,90 @@ def test_configure_env_endpoint(self, mocker):
+        assert os.environ[OTEL_EXPORTER_OTLP_ENDPOINT] == "https://otel.collector.na-01.cloud.solarwinds.com:443"
+
+    def test_configure_env_service_key_and_collector_only(self, mocker):
+        mocker.patch.dict(
+            os.environ,
+            {
+                "SW_APM_SERVICE_KEY": "foo-token:bar",
+                "SW_APM_COLLECTOR": "apm.collector.na-02.cloud.solarwinds.com",
+            },
+            clear=True
+        )
+        distro.SolarWindsDistro()._configure()
+        assert os.environ[OTEL_EXPORTER_OTLP_ENDPOINT] == "https://otel.collector.na-02.cloud.solarwinds.com:443"
+
+    def test_configure_env_service_key_and_collector_only_dev(self, mocker):
+        mocker.patch.dict(
+            os.environ,
+            {
+                "SW_APM_SERVICE_KEY": "foo-token:bar",
+                "SW_APM_COLLECTOR": "apm.collector.na-01.dev-ssp.solarwinds.com",
+            },
+            clear=True
+        )
+        distro.SolarWindsDistro()._configure()
+        assert os.environ[OTEL_EXPORTER_OTLP_ENDPOINT] == "https://otel.collector.na-01.dev-ssp.solarwinds.com:443"
+
+    def test_configure_env_service_key_and_collector_only_stg(self, mocker):
+        mocker.patch.dict(
+            os.environ,
+            {
+                "SW_APM_SERVICE_KEY": "foo-token:bar",
+                "SW_APM_COLLECTOR": "apm.collector.na-02.st-ssp.solarwinds.com",
+            },
+            clear=True
+        )
+        distro.SolarWindsDistro()._configure()
+        assert os.environ[OTEL_EXPORTER_OTLP_ENDPOINT] == "https://otel.collector.na-02.st-ssp.solarwinds.com:443"
+
+    def test_configure_env_service_key_and_collector_only_invalid(self, mocker):
+        mocker.patch.dict(
+            os.environ,
+            {
+                "SW_APM_SERVICE_KEY": "foo-token:bar",
+                "SW_APM_COLLECTOR": "www.google.com",
+            },
+            clear=True
+        )
+        distro.SolarWindsDistro()._configure()
+        assert os.environ[OTEL_EXPORTER_OTLP_ENDPOINT] == "https://otel.collector.na-01.cloud.solarwinds.com:443"
+
+    def test_configure_env_service_key_and_collector_only_invalid_otel(self, mocker):
+        mocker.patch.dict(
+            os.environ,
+            {
+                "SW_APM_SERVICE_KEY": "foo-token:bar",
+                "SW_APM_COLLECTOR": "otel.collector.na-02.cloud.solarwinds.com",
+            },
+            clear=True
+        )
+        distro.SolarWindsDistro()._configure()
+        assert os.environ[OTEL_EXPORTER_OTLP_ENDPOINT] == "https://otel.collector.na-01.cloud.solarwinds.com:443"
+
+    def test_configure_env_service_key_and_collector_and_otel_exporter(self, mocker):
+        mocker.patch.dict(
+            os.environ,
+            {
+                "SW_APM_SERVICE_KEY": "foo-token:bar",
+                "SW_APM_COLLECTOR": "apm.collector.na-02.cloud.solarwinds.com",
+                "OTEL_EXPORTER_OTLP_ENDPOINT": "https://apm.collector.na-03.cloud.solarwinds.com:443",
```

Comment:
this is a good test that we respect explict OTEL exporter config as highest precedence. my nit is the value could be something that does not imply the apm.collector actually supports OTLP.

Reply:
Sure. Done